File: src/lib/api/audio.ts
import { invoke } from "@tauri-apps/api/core";
import type { AudioDeviceInfo, SystemAudioState } from "./types";

export const audioApi = {
  getState: () => invoke<SystemAudioState>("get_system_audio_state"),
  listDevices: () => invoke<AudioDeviceInfo[]>("list_audio_output_devices"),
  setVolume: (volumePercent: number) =>
    invoke<void>("set_system_volume", { volumePercent: Math.round(volumePercent) }),
  setDefaultDevice: (deviceId: string) =>
    invoke<void>("set_default_audio_output", { deviceId }),
};

File: src-tauri/src/commands/audio.rs
use crate::error::AppResult;
use crate::models::{AudioDeviceInfo, SystemAudioState};
use crate::services;

#[tauri::command]
pub fn get_system_audio_state() -> AppResult<SystemAudioState> {
    services::get_system_audio_state()
}

#[tauri::command]
pub fn list_audio_output_devices() -> AppResult<Vec<AudioDeviceInfo>> {
    services::list_audio_output_devices()
}

#[tauri::command]
pub fn set_system_volume(volume_percent: u8) -> AppResult<()> {
    services::set_system_volume(volume_percent)
}

#[tauri::command]
pub fn set_default_audio_output(device_id: String) -> AppResult<()> {
    services::set_default_audio_output(&device_id)
}
